.github/workflows/test.yaml

```diff
@@ -3,11 +3,14 @@ on:
   pull_request:
     branches: [ "master", "release/**" ]
   push:
-    branches: [ "release/**" ]
+    # we trigger runs on master branch, but we do not run spread on master 
+    # branch, the master branch runs are just for unit tests + codecov.io
+    branches: [ "master","release/**" ]
 
 jobs:
   snap-builds:
     runs-on: ubuntu-20.04
+    if: ${{ github.event_name != 'push' && github.ref != 'refs/heads/master' }}
```

Comment:
Isn't just the test `github.ref != 'refs/heads/master'` enough? It's not clear why you are also checking the event type, and whether that `&&` should be a `||` instead.

Reply:
We still want the spread tests to run on push to release branches AIUI

Reply:
Though perhaps that was unintentional when we switched to GitHub Actions, @mvo5 do you recall if it's intentional to run the spread tests on pushes to the release branches? I think it is, as it aids @sergiocazzolato when doing beta validation to be able to point to successful spread runs on the release/2.51 branch for 2.51 SRU validation I think

Reply:
> We still want the spread tests to run on push to release branches AIUI

But if the event is a push, the condition will be false, right? Or am I reading it wrong?

Reply:
Ah I see what you mean. If we use the condition `github.ref != 'refs/head/master'`, then it will not run when we file PR's to master like this one. The situations are essentially a matrix, where the unit tests run everywhere and the other jobs run like this:

| event ↓ / branch → | master | release/** |
| ---- | ---- | ---- |
| push | :x:   | :heavy_check_mark: |
| pr | :heavy_check_mark: | :heavy_check_mark: |

does  that help?

Reply:
It helps in understanding what you want to achieve, but it still feels to me that the condition then should be
```
    if: ${{ github.event_name != 'push' || github.ref != 'refs/heads/master' }}
```

Reply:
Ah yes I'm sorry you're exactly right the condition as I have it won't run on pushes to release. I will file a new PR with the fix.